include/aikido/planner/kinodynamic/KinodynamicTimer.hpp

```diff
@@ -0,0 +1,139 @@
+#ifndef AIKIDO_PLANNER_KINODYNAMIC_KINODYNAMICTIMER_HPP_
+#define AIKIDO_PLANNER_KINODYNAMIC_KINODYNAMICTIMER_HPP_
+
+#include <Eigen/Dense>
+#include "aikido/planner/TrajectoryPostProcessor.hpp"
+#include "aikido/trajectory/Interpolated.hpp"
+#include "aikido/trajectory/Spline.hpp"
+
+namespace aikido {
+namespace planner {
+namespace kinodynamic {
+
+/// Computes the time-optimal timing of a trajectory consisting of a sequence
+/// Geodesic interpolations between states under velocity and acceleration
+/// bounds. The output is a parabolic spline, encoded in cubic polynomials.
+/// It firstly preprocesses a non-differentiable path to a differentiable one
+/// by adding circular blends; and then \b exactly follows the preprocessed
+/// path.
+///
+/// The output trajectory consists of a sequence of trapezoidal velocity
+/// profiles that implement bang-bang control. This function curently only
+/// supports \c RealVector, \c SO2, and compound state spaces of those types.
+/// Additionally, this function requires that \c inputTrajectory to be
+/// interpolated using a \c GeodesicInterpolator.
```

Comment:
Why does the input trajectory need to have been interpolated using `GeodesicInterpolator`? Why the explicit mention? Just curious. Any sequence of waypoints should be fine right?

Reply:
I think this is necessary. The interpolation implies the shape of a path given a sequence of waypoints. If you choose a different interpolator, the shape of a path will be different (length might also be changed). The timing algorithm works on the assumption of the geodesic interpolation.